admin/services/pupil-census.service.js

```diff
@@ -97,11 +98,35 @@ pupilCensusService.getUploadedFile = async () => {
   if (!jobStatus) {
     throw new Error(`There is no job status for job status id ${jobStatusId}`)
   }
-  const outcome = jobStatus.jobStatusCode === 'CWR'
-    ? `${jobStatus.description} : ${pupilCensus.errorOutput}` : `${jobStatus.description} : ${pupilCensus.jobOutput}`
+  const completedWithErrors = 'CWR'
+  const deleted = 'DEL'
+  let outcome
+  if (jobStatus.jobStatusCode === completedWithErrors) {
+    outcome = pupilCensus.errorOutput
+  } else if (jobStatus.jobStatusCode === deleted) {
+    outcome = jobStatus.description
+  } else {
+    outcome = `${jobStatus.description} : ${pupilCensus.jobOutput}`
+  }
   pupilCensus.csvName = pupilCensus.jobInput
   pupilCensus.outcome = outcome
+  pupilCensus.jobStatusCode = jobStatus.jobStatusCode
```

Comment:
Is there any chance that `jobStatusCode` is tainted?  Or is it **guaranteed** to be a correct value?

Reply:
It's been populated directly from the db record so I believe it's guaranteed.

Reply:
that's fine then